src/streams.rs

```diff
@@ -0,0 +1,695 @@
+                            })
+                            .collect();
+                        StreamKey { key, ids }
+                    })
+                    .collect::<Vec<StreamKey>>()
+            })
+            .collect();
+        Ok(StreamReadReply { keys })
+    }
+}
+
+impl FromRedisValue for StreamRangeReply {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        let rows: Vec<HashMap<String, HashMap<String, Value>>> = from_redis_value(v)?;
+        let ids: Vec<StreamId> = rows
+            .into_iter()
+            .flat_map(|row| {
+                row.into_iter()
+                    .map(|(id, map)| StreamId { id, map })
+                    .collect::<Vec<StreamId>>()
+            })
+            .collect();
+        Ok(StreamRangeReply { ids })
+    }
+}
+
+impl FromRedisValue for StreamClaimReply {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        let rows: Vec<HashMap<String, HashMap<String, Value>>> = from_redis_value(v)?;
+        let ids: Vec<StreamId> = rows
+            .into_iter()
+            .flat_map(|row| {
+                row.into_iter()
+                    .map(|(id, map)| StreamId { id, map })
+                    .collect::<Vec<StreamId>>()
+            })
+            .collect();
+        Ok(StreamClaimReply { ids })
+    }
+}
+
+type SPRInner = (
+    usize,
+    Option<String>,
+    Option<String>,
+    Vec<Option<(String, String)>>,
+);
+impl FromRedisValue for StreamPendingReply {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        let (count, start, end, consumer_data): SPRInner = from_redis_value(v)?;
+
+        if count == 0 {
+            Ok(StreamPendingReply::Empty)
+        } else {
+            let mut result = StreamPendingData::default();
+
+            let start_id = start.ok_or_else(|| {
+                Error::new(
+                    ErrorKind::Other,
+                    "IllegalState: Non-zero pending expects start id",
+                )
+            })?;
+
+            let end_id = end.ok_or_else(|| {
+                Error::new(
+                    ErrorKind::Other,
+                    "IllegalState: Non-zero pending expects end id",
+                )
+            })?;
+
+            result.count = count;
+            result.start_id = start_id;
+            result.end_id = end_id;
+
+            for cd in consumer_data {
+                if let Some((name, pending)) = cd {
+                    let mut info = StreamInfoConsumer::default();
+                    info.name = name;
+                    if let Ok(v) = pending.parse::<usize>() {
+                        info.pending = v;
+                    }
+                    result.consumers.push(info);
+                }
+            }
+
+            Ok(StreamPendingReply::Data(result))
+        }
+    }
+}
+
+impl FromRedisValue for StreamPendingCountReply {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
```

Comment:
This is the area that was previously under discussion.  Now manually matches the `&Value`.